### ISSUE TITLE
file search: DELETE causes random search result tree nodes to expand

### ISSUE THREAD

**Issue by reto-hoehener** [eclipse-platform/eclipse.platform.ui#940]
To reproduce:
- use file search to search e.g. for *.java files
- using the search result tree view, select a few projects in the search results and hit DELETE
- now all subsequent delete or navigation actions in the search result view jump to seemingly random nodes in the tree

It looks like the tree view is not updated to reflect the deletions, which cause the next actions to jump to wrong positions and expand "random" nodes.

Happening after switching from 2022-06 to 2023-06.

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/13407181/82524ff0-4962-43bc-979c-806b5d6c7689)

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/13407181/5ec7b109-4e19-46df-b345-80059e532ae1)

**Comment by reto-hoehener** [comment]
Still happening in Eclipse 2023-09. Searching for files in my workspace and working through the results one by one is a very important process for me. Please increase priority for this. It is immediately and easily reproducible. Please let me know if you need any more info to reproduce.

**Comment by reto-hoehener** [comment]
Note: Leave the 'Containing text'-search field blank, only use 'File name patterns'.

I suspect that the tree selection requires actual 'content'-search-hits within the files to work correctly. If you only search for filename patterns, the whole result tree model stops to work correctly.

**Comment by merks** [comment]
I was about to say that I can't reproduce it until you mentioned the blank thing, and that's indeed totally screwed up.  I don't know what delete is even doing other than screwing up the navigation.  So your own purpose here is to have a list of all *.java files, not to do any kind of "meaningful" search...

**Comment by reto-hoehener** [comment]
@merks no the *.java search was just for reproduction purposes. For example I need to search the 170+ projects in my workspace for files whose names match `*_Foo*.java` or `*_Bar*.properties` or `*XYZ*.config` or `*Bla*.settings` (etc.) and then proceed to do something with them (maybe I need to edit them, or just read or check them or whatever).

**Comment by merks** [comment]
My point is that the search string being empty is necessary to reproduce the problem, right?  I do understand that having a list of files that match the file name pattern could be useful.  It's important to be clear on exactly what's needed to reproduce the problem and I think it only happens for an empty search string...

Would you be interested in trying to contribute a fix?

**Comment by reto-hoehener** [comment]
Yes I believe that is correct. This used to work. I considered using other tools, but I really need my IDE to jump to the file and view/edit it quickly. I edited my reply for irritated tone, please forgive me, I get scared when the most core functions of my IDE (searching / replacing / refactoring) stop working reliably.  I don't have resources for contributing sorry, please consider my spending time reporting my contribution (I know noone else bothers in my team and just switches to other tools).
